src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -41,8 +41,9 @@ class PaginationExtension implements QueryResultExtensionInterface
     private $enabledParameterName;
     private $itemsPerPageParameterName;
     private $maximumItemPerPage;
+    private $maximumItemPerPageEnabled;
 
-    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $pageParameterName = 'page', string $enabledParameterName = 'pagination', string $itemsPerPageParameterName = 'itemsPerPage', int $maximumItemPerPage = 300)
+    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $pageParameterName = 'page', string $enabledParameterName = 'pagination', string $itemsPerPageParameterName = 'itemsPerPage', int $maximumItemPerPage = 300, bool $maximumItemPerPageEnabled = false)
```

Comment:
Can't you just change the signature to `int $maximumItemPerPage = null` (and it will be disabled when `null` is passed).